slang.h

```diff
@@ -1841,7 +1842,11 @@ extern "C"
         SLANG_RESOURCE_UNKNOWN              = 0x08,
         SLANG_ACCELERATION_STRUCTURE        = 0x09,
 
-        SLANG_RESOURCE_EXT_SHAPE_MASK       = 0xF0,
+        //SLANG_RESOURCE_EXT_SHAPE_MASK       = 0xF0,
+
+        SLANG_TEXTURE_FLAG_MASK             = 0xF0,
+
+        SLANG_TEXTURE_FEEDBACK_FLAG         = 0x10,
```

Comment:
Hmm... at the very least it seems like we need to keep the old `SLANG_RESOURCE_EXT_SHAPE_MASK` enumeration so that existing source can continue working.

I'm a bit confused about the rename, though. Whether or not a texture is a feedback texture isn't really part of its "shape," but once we've committed to storing the relevant bit inside `SlangResourceShape` we are kind of stuck with pretending like it is, right?

The idea of "shape" here was that you had the "base shape" (2D, 3D, etc.) and then the "extended shape" included any modifiers that might lead to it requiring new texture coordinate values on lookups (which both array-ness and multisample-ness do).

Does feedback-ness of a texture need to be exposed through reflection? If it does, does it need to be exposed as part of the "shape" of a texture?

Reply:
I think I'm convinced that this solution is the best one available right now, but I'd encourage *not* renaming the existing mask enumeration (both returning it to the original name *and* removing the new name).

We probably need something cleaner for exposing aspects of resource types that aren't conceptually part of the "shape," but that seems best to do later as part of an overhaul of this logic (e.g., we are probably close to the point where squeezing all this into one byte turns out to have been a short-sighted optimization on my part).

Reply:
I reinstated SLANG_RESOURCE_EXT_SHAPE_MASK, and see there is argument for it remaining if just for compatibility.  I changed it because it's name doesn't seem to match it's use.